### DISCUSSION TITLE
Defining Mill builds in pure Scala?

### DISCUSSION THREAD

**Discussion by itegebo** [com-lihaoyi/mill#1368] [Q&A] [answered]
I'm looking to be able to use Mill in a team where we aren't familiar with Ammonite Script files (and wouldn't otherwise have much of a need for them).  The larger build system disallows network access and we'd be supplying our own dependency resolution.  It doesn't seem like Ammonite Script files add any value and actually seem to add [unnecessary complications in this case](https://github.com/com-lihaoyi/mill/discussions/1293).

Would someone be able to comment on how difficult it might be to provide a way to define Mill builds in pure Scala?

**Comment by lefou** [comment, marked as answer]
Who is supposed to compile your pure Scala file and feed the right classpath to the compiler? That is what you get when using ammonite / mill scripts. 
Once you have your file compiled it might be pretty easy to load it in either some stub `build.sc` or via an external plugin.

Without any knowledge about your external dependency resolution process, you could adapt/override `CoursierModule` to use your external system.

In essence, when you look at external mill plugins (e.g. https://github.com/lefou/mill-vcs-version), you see what is needed to provide and builds your build definitions outside of mill. All you need in the `build.sc` is to load the resulting jar.
